Repository/ProductRepository.cs

```diff
@@ -0,0 +1,71 @@
+﻿using EFCoreServices.Models;
+using Microsoft.EntityFrameworkCore;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace EFCoreServices.Repository
+{
+    public class ProductRepository : IProductRepository
+    {
+        ProductDBContext db;
+        public ProductRepository(ProductDBContext _DbContext)
+        {
+            db = _DbContext;
+        }
+
+        public async Task<int> AddProduct(Product prod)
+        {
+            if(db != null)
+            {
+                await db.Products.AddAsync(prod);
+                await db.SaveChangesAsync();
+
+            }
+            return 0;
```

Comment:
try to always return the added product and return that to the end user. 
also don't directly deal with model in your controllers 
you need DTO for the same. 
create request and response DTO for product along with its model. 
and map values based on where it is going